### PR TITLE
feat: CP support Input.TextArea allowClear、autoComplete、className、classNames、style、styles

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- #46889

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | CP support Input.TextArea `allowClear` <br /> CP support Input.TextArea `autoComplete` <br /> CP support Input.TextArea `className`、`style` <br /> CP support Input.TextArea `classNames`、`styles` |
| 🇨🇳 Chinese | CP 支持 Input.TextArea 组件 `allowClear` 属性 <br /> CP 支持 Input.TextArea 组件 `autoComplete` 属性 <br /> CP 支持 Input.TextArea 组件 `className`、`style` 属性 <br /> CP 支持 Input.TextArea 组件 `classNames`、`styles` 属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/TextArea.tsx [hunk 1]
```diff
@@ -61,7 +60,7 @@ const TextArea = forwardRef<TextAreaRef, TextAreaProps>((props, ref) => {
     deprecated(!('bordered' in props), 'bordered', 'variant');
   }
 
-  const { getPrefixCls, direction } = React.useContext(ConfigContext);
+  const { getPrefixCls, direction, input } = React.useContext(ConfigContext);
```
Comment (MadCcc) [inline components/input/TextArea.tsx:63]: TextArea 要单独开一个属性 
https://github.com/ant-design/ant-design/discussions/46170

File: components/config-provider/context.ts [hunk 2]
```diff
@@ -102,6 +102,11 @@ export type BadgeConfig = ComponentStyleConfig & Pick<BadgeProps, 'classNames' |
 
 export type ButtonConfig = ComponentStyleConfig & Pick<ButtonProps, 'classNames' | 'styles'>;
 
+export type InputConfig = ComponentStyleConfig &
+  Pick<InputProps, 'autoComplete' | 'classNames' | 'styles' | 'allowClear'>;
+
+export type TextAreaConfig = ComponentStyleConfig & Pick<InputProps, 'allowClear'>;
```
Comment (MadCcc) [inline components/config-provider/context.ts:108]: TextArea 有 TextAreaProps，不要和 Input 混起来

Comment (li-jia-nan) [reply components/config-provider/context.ts:108]: 我还在写……这个我先标记哥 WIP 吧

File: components/config-provider/__tests__/style.test.tsx [hunk 3]
```diff
@@ -515,6 +515,58 @@ describe('ConfigProvider support style and className props', () => {
...
+              color: 'red',
+            },
+          },
+          allowClear: {
+            clearIcon: <span className="cp-test-icon">cp-test-icon</span>,
+          },
+        }}
+      >
+        <Input.TextArea
+          autoComplete="test-autocomplete"
```
Comment (MadCcc) [inline components/config-provider/__tests__/style.test.tsx:542]: 测一下 CP 上配置 autocomplete 以及两边都有的用例

**Comment by MadCcc** [comment]
changelog 合起来写就行，不要加 feat: 前缀，最后是要可以直接用作 changelog 的

File: components/config-provider/__tests__/style.test.tsx (hunk 3 above)
Comment (li-jia-nan) [reply components/config-provider/__tests__/style.test.tsx:557]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**